packages/engine/src/ecs/classes/Scene.ts

```diff
@@ -23,34 +23,119 @@ All portions of the code written by the Ethereal Engine team are Copyright © 20
-import { defineState, getMutableState } from '@etherealengine/hyperflux'
+import { Color, FileLoader, MathUtils, Texture } from 'three'
 
+import { EntityUUID } from '@etherealengine/common/src/interfaces/EntityUUID'
+import { SceneJson } from '@etherealengine/common/src/interfaces/SceneInterface'
+import { defineState, getMutableState, none } from '@etherealengine/hyperflux'
+import { AssetLoader } from '../../assets/classes/AssetLoader'
+import { NameComponent } from '../../scene/components/NameComponent'
+import { SceneTagComponent } from '../../scene/components/SceneTagComponent'
+import { UUIDComponent } from '../../scene/components/UUIDComponent'
+import { VisibleComponent } from '../../scene/components/VisibleComponent'
+import { TransformComponent } from '../../transform/components/TransformComponent'
+import { setComponent } from '../functions/ComponentFunctions'
+import { createEntity } from '../functions/EntityFunctions'
+import { EntityTreeComponent } from '../functions/EntityTree'
 import { Engine } from './Engine'
-import { UndefinedEntity } from './Entity'
+
+export interface StagedScene {
+  data?: SceneJson
+  load: boolean
+  loadProgress: {
+    textures: number
+    geometries: number
+    rigidbodies: number
+  }
+}
 
 export const SceneState = defineState({
   name: 'SceneState',
-  initial: () => ({
-    sceneData: null as SceneData | null,
-    sceneEntity: UndefinedEntity,
-    /** @todo support multiple scenes */
-    // sceneEntities: {} as Record<string /* SceneID */, EntityUUID>,
-    background: null as null | Color | Texture
-  })
+
+  initial: () => {
+    const sceneEntity = createEntity()
+    setComponent(sceneEntity, NameComponent, 'scenes')
+    setComponent(sceneEntity, VisibleComponent, true)
+    setComponent(sceneEntity, UUIDComponent, MathUtils.generateUUID() as EntityUUID)
+    setComponent(sceneEntity, SceneTagComponent, true)
+    setComponent(sceneEntity, TransformComponent)
+    setComponent(sceneEntity, EntityTreeComponent, { parentEntity: null })
+
+    return {
+      sceneEntity,
+      scenes: {} as Record<string, StagedScene>,
+      background: null as null | Color | Texture
+    }
+  },
+
+  fetchScene: (url: string) => {
+    const scene = getMutableState(SceneState).scenes[url]
+    if (!scene.value) {
+      scene.set({
+        data: undefined,
+        loadProgress: {
+          textures: 0,
+          geometries: 0,
+          rigidbodies: 0
+        },
+        load: false
+      })
+
+      const resolvedURL = AssetLoader.getAbsolutePath(url)
+      const fileLoader = new FileLoader()
+      fileLoader.load(resolvedURL, (data) => {})
+      Engine.instance.api
+        .service('scene')
+        .get({ url, metadataOnly: null }, {})
+        .then((sceneData) => {
+          if (scene.value) scene.data.set(sceneData.data.scene)
+        })
+    }
+    return scene
+  },
+
+  loadScene: (url: string) => {},
```

Comment:
we call this in set scene, are we planning to define this function?

Reply:
this PR is still a WIP